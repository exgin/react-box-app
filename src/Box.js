import React from 'react';
import './Box.css';

const Box = ({ id, removeBox, color, width, height }) => {
  const remove = () => {
    removeBox(id);
  };

  return (
    <div className='Box'>
      <p style={{ backgroundColor: color, height: `${height}px`, width: `${width}px` }}>box</p>
      <button className='Box-btn' onClick={remove}>
        Remove {color}
      </button>
    </div>
  );
};

export default Box;
