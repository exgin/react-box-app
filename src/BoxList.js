import React, { useState } from 'react';
import Box from './Box';
import NewBoxForm from './NewBoxForm';

const BoxList = () => {
  const [boxes, setBox] = useState([]);

  const addBox = (newBox) => {
    setBox((boxes) => [...boxes, newBox]);
  };

  const removeBox = (id) => {
    setBox((boxes) => boxes.filter((box) => box.id !== id));
  };

  const allBoxComps = boxes.map((box) => <Box key={box.id} id={box.id} height={box.height} width={box.height} color={box.color} removeBox={removeBox} />);

  return (
    <div>
      <h1>Boxes</h1>
      <>
        <NewBoxForm addBox={addBox} />
      </>

      <>{allBoxComps}</>
    </div>
  );
};

export default BoxList;
