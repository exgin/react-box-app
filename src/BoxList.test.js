import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import BoxList from './BoxList';
import Box from './Box';

it('should render successfully', function () {
  render(<BoxList />);
});

it('should match snapshot', function () {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

it('should contain all form inputs', function () {
  const { getByLabelText } = render(<BoxList />);
  const color = getByLabelText('Color:');
  const height = getByLabelText('Height:');
  const width = getByLabelText('Width:');

  expect(color).toBeInTheDocument();
  expect(height).toBeInTheDocument();
  expect(width).toBeInTheDocument();
});

it('should make a new box', function () {
  const { getByLabelText, getByText } = render(<BoxList />);
  const color = getByLabelText('Color:');
  const height = getByLabelText('Height:');
  const width = getByLabelText('Width:');

  fireEvent.change(color, { target: { value: 'blue' } });
  fireEvent.change(height, { target: { value: '50px' } });
  fireEvent.change(width, { target: { value: '50px' } });

  const button = getByText('Add a new Box');
  fireEvent.click(button);

  //   find box & remove btn in the dom
  const box = getByText('box');
  expect(box).toBeInTheDocument();

  const rmBtn = getByText('Remove', { exact: false });
  expect(rmBtn).toBeInTheDocument();
});
