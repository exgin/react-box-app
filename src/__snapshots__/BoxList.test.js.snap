// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Boxes
    </h1>
    <form>
      <label
        for="color"
      >
         Color: 
      </label>
      <input
        id="color"
        name="color"
        value=""
      />
      <label
        for="height"
      >
         Height: 
      </label>
      <input
        id="height"
        name="height"
        value=""
      />
      <label
        for="width"
      >
         Width: 
      </label>
      <input
        id="width"
        name="width"
        value=""
      />
      <button>
        Add a new Box
      </button>
    </form>
  </div>
</DocumentFragment>
`;
